configure.sh

```diff
@@ -124,6 +133,33 @@ case $(uname -s) in
         BUILD_VIRTIO="yes"
         BUILD_MUEN="yes"
         ;;
+    OpenBSD)
+        # On OpenBSD/clang we use -nostdlibinc which gives us access to the
+        # clang-provided headers for compiler instrinsics. We copy the rest
+        # (std*.h, cdefs.h and their dependencies) from the host.
+        cc_is_clang || die "Only 'clang' is supported on OpenBSD"
+        [ "${TARGET_ARCH}" = "x86_64" ] ||
+            die "Only 'x86_64' is supported on OpenBSD"
+        ld_is_lld || die "Only 'ld.lld' is supported on OpenBSD"
```

Comment:
Are there other versions of `ld` (that would be in a standard system $PATH) on OpenBSD?

Reply:
OpenBSD's default ld is GNU ld version 2.17. which is just too old for the Solo5's linker scripts. OpenBSD's amd64 port is compiled with clang but linked with ld, but always has ld.lld available.

Reply:
What do you mean by "always has ld.lld available"? Does this require manually using `LD=ld.lld` in various places to make the build work?

Reply:
yes, before working on solo5 and MirageOS i run `export LD=ld.lld` in my shell. and everything seems to build correctly.

Reply:
Hmm. We should have a way to make this work w/o relying on that step. I'll go look back at the dance we did ages ago when we were still depending on `/usr/local/bin/ld` for the FreeBSD port and get back to you.

Reply:
where you thinking something along the lines of [encode ld.lld as a default LD, similar to FreeBSD a while ago](https://github.com/adamsteen/solo5/commit/c69f83462c3023022ff525ba796cb5e3f46c5831) like what was removed in [remove LD hacks](https://github.com/Solo5/solo5/commit/a39d7f69e9044a6bd90571fe1d6391a347591e46#diff-735003ab6f5797c8b8bdecbbbb9322a8)?

Reply:
Yeah, that's it. Unfortunately that requires putting it back in downstream (Mirage) as well. Could you find out if there are any plans (e.g. by the next OpenBSD release) to make `ld.lld` the default linker? If that were the case then I'd rather just wait until that happens.

Reply:
I doubt ld.lld will be made the default linker any time soon, due to [Bug 30815 - linking OpenBSD/amd64 kernel](https://bugs.llvm.org/show_bug.cgi?id=30815) (The linked kernel doesn't boot.) but i have sent an [email](https://marc.info/?l=openbsd-misc&m=151330492417254&w=2) any ways to see what the projects plans are.